eng/scripts/Invoke-GenerateAndBuildV2.ps1

```diff
@@ -113,41 +113,35 @@ if ($inputFileToGen) {
         $typespecFolder = Resolve-Path (Join-Path $swaggerDir $typespecRelativeFolder)
-        $processScript = Resolve-Path (Join-Path $sdkPath "eng/common/scripts" "TypeSpec-Project-Process.ps1")
-        $sdkProjectFolders = Get-ChildItem -Path (Join-Path $sdkPath "sdk") -Depth 1 -Directory | Select-Object -ExpandProperty FullName
-
-        # Invoke Process script. SkipSyncAndGenerate only when it's not a new SDK project
-        $sdkProjectFolder = & $processScript $typespecFolder $commitid $repoHttpsUrl -SkipSyncAndGenerate
+        $tspConfigFile = Resolve-Path (Join-Path $typespecFolder "tspconfig.yaml")
+        $sdkProjectFolder = GetSDKProjectFolder -typespecConfigurationFile $tspConfigFile -sdkRepoRoot $sdkPath
+        $sdkAutorestConfigFile = Join-path $sdkProjectFolder "src" "autorest.md"
+        if (Test-Path -Path $sdkAutorestConfigFile) {
+            Write-Host "remove $sdkAutorestConfigFile for sdk from typespec."
+            Remove-Item -Path $sdkAutorestConfigFile
+        }
+        $repo = $repoHttpsUrl -replace "https://github.com/", ""
+        $tspclientCommand = "npx tsp-client init --tsp-config $tspConfigFile --repo $repo --commit $commitid"
+        Write-Host $tspclientCommand
+        Invoke-Expression $tspclientCommand
         if ($LASTEXITCODE) {
           # If Process script call fails, then return with failure to CI and don't need to call GeneratePackage
+          Write-Error "Failed to generate typespec project. Exit code: $LASTEXITCODE"
           $generatedSDKPackages.Add(@{
             result = "failed";
             path=@("");
           })
         } else {
             $relativeSdkPath = Resolve-Path $sdkProjectFolder -Relative
-            if ($sdkProjectFolders -contains $sdkProjectFolder) {
-              # Existed SDK project case, needs to generate code
-              GeneratePackage `
-              -projectFolder $sdkProjectFolder `
-              -sdkRootPath $sdkPath `
-              -path $relativeSdkPath `
-              -downloadUrlPrefix $downloadUrlPrefix `
-              -serviceType "data-plane" `
-              -generatedSDKPackages $generatedSDKPackages `
-              -specRepoRoot $swaggerDir
-            } else {
-              # New SDK project case, code is already generated by emitter. So, skip code generation
-              GeneratePackage `
-              -projectFolder $sdkProjectFolder `
-              -sdkRootPath $sdkPath `
-              -path $relativeSdkPath `
-              -downloadUrlPrefix $downloadUrlPrefix `
-              -serviceType "data-plane" `
-              -skipGenerate `
-              -generatedSDKPackages $generatedSDKPackages `
-              -specRepoRoot $swaggerDir
-            }
+            GeneratePackage `
+            -projectFolder $sdkProjectFolder `
+            -sdkRootPath $sdkPath `
+            -path $relativeSdkPath `
+            -downloadUrlPrefix $downloadUrlPrefix `
+            -serviceType "data-plane" `
```

Comment:
Should this function process both `data plane` and `mgmt. plane` generation from TypeSpec?

Reply:
Yes. if mgmt-plane come from TypeSpec. If a SDK from TypeSpec, there will be change item in `$inputJson.relatedTypeSpecProjectFolder` (inputJsonFile from Pipeline framework)

Reply:
Then here should not hard code "data-plane" as `serviceType`?

Reply:
resolved